flink-tests/src/test/java/org/apache/flink/test/plugin/jar/failurelistener/TestFailureListener.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.test.plugin.jar.failurelistener;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.core.failurelistener.FailureListener;
+import org.apache.flink.metrics.MetricGroup;
+
+/** Implementation of {@link FailureListener} for plugin loading test. */
+public class TestFailureListener implements FailureListener {
+
+    @Override
+    public void init(JobID jobID, String jobName, MetricGroup metricGroup) {}
```

Comment:
I don't think the init() method is really needed. All this information can be passed into the factory.

Reply:
Resolved